java-checks/src/main/java/org/sonar/java/checks/spring/SpringEndpointsCheck.java

```diff
@@ -36,29 +35,26 @@
 
   @Override
   public List<Tree.Kind> nodesToVisit() {
-    return Collections.singletonList(Tree.Kind.ANNOTATION);
+    return Collections.singletonList(Tree.Kind.METHOD);
   }
 
   @Override
   public void visitNode(Tree tree) {
-    AnnotationTree annotationTree = (AnnotationTree) tree;
-    if (isSpringWebHandler(annotationTree)) {
-      findParentMethod(annotationTree).ifPresent(annotatedMethod ->
-          reportIssue(annotatedMethod.simpleName(), "Review this Spring request handler"));
+    if (!hasSemantic()) {
+      return;
     }
+    MethodTree methodTree = (MethodTree) tree;
+    List<AnnotationTree> annotations = methodTree.modifiers().annotations();
+    annotations.forEach(annotationTree -> {
+      if (isSpringWebHandler(annotationTree)) {
```

Comment:
filter ?

Reply:
and to be a bit clearer : I would have a stream and filter and then for each remaining report an issue.

Reply:
yeah :)